callbacks_test.go

```diff
@@ -465,6 +488,43 @@ func Test_ExecutorCallbacks(t *testing.T) {
 				})
 			})
 
+			Convey("notifies Sidecar to drain the service before stopping the watch looper", func() {
+				dummyWatchLooper := newMockLooper()
+				exec.watchLooper = &dummyWatchLooper
+				exec.watchLooper.Loop(func() error { return nil })
+
+				sidecarDrainChan = make(chan struct{})
+				quitChan := make(chan struct{})
+				go func() {
+					exec.KillTask(&dummyTaskID)
+					// Unblock this test
+					close(quitChan)
+				}()
+
+				// Wait a bit for exec.KillTask to do some work
+				time.Sleep(100 * time.Millisecond)
```

Comment:
We should just `select` on the `quitChan` instead of sleeping, right?

Reply:
That's not going to work, because `exec.KillTask(&dummyTaskID)` calls `exec.notifyDrain()` directly, which is blocked until we run `close(sidecarDrainChan)` below. This tests does a bit of state machine modelling with channels, which is kinda' awkward, but I didn't want to refactor the rest of the code. Guess I'd have to create a mock drainer and inject it through the code to make this test simpler, but not sure if it's worth it.